### PR TITLE
Update on DG DeleteSpecialisation and List Appointment with minor update on UG, PPP

### PR DESCRIPTION
**Pull request by kimshitong**
I tried to give more detail steps on ListAppointment Command in DG, Let me know if anything doesn't go well with your mind.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -302,22 +302,23 @@ Given below is an example usage scenario and how the `ListAppointmentCommand` me
 **Step 1**: The user inputs `list_a` to list all appointments or `list_a pic\PATIENT_NRIC dic\DOCTOR_NRIC` to filter appointments.
 * The `list_a` command triggers `MediConnectParser#parseCommand(String)`, which identifies the command word and calls `ListAppointmentCommandParser#parse(String)` to handle the arguments.
 
-**Step 2**: The `ListAppointmentCommandParser#parse(String)` method checks for the presence of optional flags like `dic\` for doctor NRIC and `pic\` for patient NRIC. Based on these, it creates appropriate `Predicate` objects.
+**Step 2**: The `ListAppointmentCommandParser#parse(String)` method checks for the presence of optional flags like `dic\` for doctor NRIC and `pic\` for patient NRIC.
+ * Based on the presence of the doctor/patient's NRIC, it will affect whether their predicate `AppointmentEqualDoctorNricPredicate`/`AppointmentEqualPatientNricPredicate` is `PREDICATE_SHOW_ALL_DOCTORS`/`PREDICATE_SHOW_ALL_PATIENTS` or the predicate of the specified doctor/patient's NRIC.
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:306]: maybe can say `based on the presence of the doctor/patient NRIC flags`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -302,22 +302,23 @@ Given below is an example usage scenario and how the `ListAppointmentCommand` me
...
+ * A new `ListAppointmentCommand` instance is created using the `AppointmentFilterByNricPredicate` that is constructed with `AppointmentEqualDoctorNricPredicate` and `AppointmentEqualPatientNricPredicate`.
 
-**Step 3**: A new `ListAppointmentCommand` instance is created using the `Predicate` object(s).
+**Step 3**: The created `ListAppointmentCommand` instance is returned to `LogicManager`, and its execute method is called.
+ * `ListAppointmentCommand#execute(Model)` then calls `Model#updateFilteredAppointmentList(Predicate<Appointment>)` with the predicate `AppointmentFilterByNricPredicate`.
+ * The `FilteredAppointmentList` is updated to show all doctors by calling `ObservableList#setPredicate(Predicate<Appointment>)`.
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:311]: shld call Model#setPredicate(Predicate<Appointment> instead

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (kimshitong) [reply docs/DeveloperGuide.md:306]: ok, thanks!

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (kimshitong) [reply docs/DeveloperGuide.md:311]: Model#updateFilteredAppointmentList(Predicate<Appointment>) instead.

Thanks!

Comment (kimshitong) [reply docs/DeveloperGuide.md:311]: Gets! Thanks

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -302,22 +302,23 @@ Given below is an example usage scenario and how the `ListAppointmentCommand` me
...
+**Step 2**: The `ListAppointmentCommandParser#parse(String)` method checks for the presence of optional flags like `dic\` for doctor NRIC and `pic\` for patient NRIC.
+ * Based on the presence of the doctor/patient NRIC flags, it will affect whether their predicate `AppointmentEqualDoctorNricPredicate`/`AppointmentEqualPatientNricPredicate` is `PREDICATE_SHOW_ALL_DOCTORS`/`PREDICATE_SHOW_ALL_PATIENTS` or the predicate of the specified doctor/patient's NRIC.
+ * A new `ListAppointmentCommand` instance is created using the `AppointmentFilterByNricPredicate` that is constructed with `AppointmentEqualDoctorNricPredicate` and `AppointmentEqualPatientNricPredicate`.
 
-**Step 3**: A new `ListAppointmentCommand` instance is created using the `Predicate` object(s).
+**Step 3**: The created `ListAppointmentCommand` instance is returned to `LogicManager`, and its execute method is called.
+ * `ListAppointmentCommand#execute(Model)` then calls `Model#updateFilteredAppointmentList(Predicate<Appointment>)` with the predicate `AppointmentFilterByNricPredicate`.
+ * The `FilteredAppointmentList` is updated to show all doctors by calling `Model#setPredicate(Predicate<Appointment>)`.
```
Comment (mounilsankar) [inline docs/DeveloperGuide.md:311]: Shld be Model#updateFilteredAppointmentList(Predicate<Appointment>), paiseh

Comment (kimshitong) [reply docs/DeveloperGuide.md:311]: should we specify Predicate<Appointment> or just Predicate alone is enough

Comment (mounilsankar) [reply docs/DeveloperGuide.md:311]: Predicate alone shld be fine

**Review by mounilsankar — APPROVED** [review]

**Review by mounilsankar — APPROVED** [review]

**Pull request merged**